include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,548 @@
+    class sentinel
+    {
+        using difference_type =
+            std::common_type_t<std::ranges::range_difference_t<__internal::__maybe_const<_Const, _Views>>...>;
+
+        template <typename... Sentinels>
+        constexpr explicit sentinel(Sentinels... sentinels) : __end(std::move(sentinels)...)
+        {
+        }
+
+      public:
+        sentinel() = default;
+        constexpr sentinel(sentinel<!_Const> i)
+            requires _Const &&
+                     (std::convertible_to<std::ranges::sentinel_t<_Views>,
+                                          std::ranges::sentinel_t<__internal::__maybe_const<_Const, _Views>>> &&
+                      ...)
+            : __end(std::move(i.__end))
+        {
+        }
+
+        template <bool _OtherConst>
+            requires(std::sentinel_for<std::ranges::sentinel_t<__internal::__maybe_const<_Const, _Views>>,
+                                       std::ranges::iterator_t<__internal::__maybe_const<_OtherConst, _Views>>> &&
+                     ...)
+        friend constexpr bool
+        operator==(const iterator<_OtherConst>& __x, const sentinel& __y)
+        {
+            return __x.__compare_with_sentinels(__y.__end, std::make_index_sequence<sizeof...(_Views)>());
+        }
+
+        template <bool _OtherConst>
+            requires(std::sized_sentinel_for<std::ranges::sentinel_t<__internal::__maybe_const<_Const, _Views>>,
+                                             std::ranges::iterator_t<__internal::__maybe_const<_OtherConst, _Views>>> &&
+                     ...)
+        friend constexpr difference_type
+        operator-(const iterator<_OtherConst>& __x, const sentinel& __y)
+        {
+            auto calc_val = [&]<std::size_t... _In>(std::index_sequence<_In...>) {
+                return std::ranges::min({difference_type(std::get<_In>(__x.__current) - std::get<_In>(__y.__end))...},
+                                        std::less{}, [](auto __a) { return std::abs(__a); });
+            };
+
+            return calc_val(std::make_index_sequence<sizeof...(_Views)>());
+        }
+
+        template <bool _OtherConst>
+            requires(std::sized_sentinel_for<std::ranges::sentinel_t<__internal::__maybe_const<_Const, _Views>>,
+                                             std::ranges::iterator_t<__internal::__maybe_const<_OtherConst, _Views>>> &&
+                     ...)
+        friend constexpr difference_type
+        operator-(const sentinel& __y, const iterator<_OtherConst>& __x)
+        {
+            return -(__x - __y);
+        }
+
+      private:
+        friend class zip_view;
+
+        __tuple_type<std::ranges::sentinel_t<__internal::__maybe_const<_Const, _Views>>...> __end;
+    }; // class sentinel
+
+    constexpr auto
+    begin()
+        requires(!(__internal::__simple_view<_Views> && ...))
+    {
+        using __iterators_type = __tuple_type<std::ranges::iterator_t<__internal::__maybe_const<false, _Views>>...>;
```

Comment:
`__iterators_type` this is already defined at the class scope,  I don't think we need it here (and in the next few functions).